### PR TITLE
Framework: Try to use Babel for all files under Webpack's control

### PR DESCRIPTION
**Pull request by gziolo**
This is an experiment to validate if we could use ES6+ in all our files.

The server side part can be easily updated to use Babel by changing `webpack.config.node.js` name.

In theory we could use `babel-node` to generate client side code, but I haven't found solution yet.

### CODE REVIEW

File: client/sections.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-const fs = require( 'fs' );
+import fs from 'fs';
```
Comment (gziolo) [inline client/sections.js]: This file is used by Webpack loader, and also is used in both server and client side code by both Webpack configs. That's why it's perfect to validate if ES6 is enabled properly.

File: server/bundler/bin/bundler.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!/usr/bin/env babel-node
```
Comment (gziolo) [inline server/bundler/bin/bundler.js]: I'm sure I did something wrong here :)

**Comment by gziolo** [comment]
Question is if we want to have it or we are happy to wait until node 8.x is out and stable? :)

It would greatly simplify all automated refactorings performed with `codemods`. Beside that it is really confusing when you discover that some of the files located in the `/client` folder need to use ES5 syntax, especially when they aren't properly documented.

**Comment by gwwar** [comment]
@gziolo unless they're running behind schedule I believe the Node 8 line should be released soon.

**Comment by gziolo** [comment]
@gwwar Active LTS Start: 2017-10-01. Isn't as soon as I thought. At least 4 more months. Another issue we could have is that Node code still would differ from what Babel offers. I would say we should try to use one approach on the whole codebase.

File: server/bundler/bin/bundler.js (hunk 2 above)
Comment (blowery) [inline server/bundler/bin/bundler.js]: well, that's looking for babel-node in the shell's path. Do we want to use a babel-node that we install via a dev dependency? Or rely on the user having babel installed (and the right version) globally?

Comment (gziolo) [inline server/bundler/bin/bundler.js]: We should rather provide `babel-node` through `node_modules`.

**Comment by gziolo** [comment]
It looks like it will take at least a year until `import` and `export` works with Node. See https://medium.com/the-node-js-collection/an-update-on-es6-modules-in-node-js-42c958b890c.

**Comment by ehg** [comment]
We should be careful with this, especially with our recent RAM issues. I agree with the reasoning, but it makes me uneasy :/

https://babeljs.io/docs/usage/cli/#babel-node

**Comment by gziolo** [comment]
>  I agree with the reasoning, but it makes me uneasy :/
> 
> https://babeljs.io/docs/usage/cli/#babel-node

We use transpiled code on production, so it shouldn't be an issue. Anyway, it can indeed slow down build process.

Closing this for now as it seems tricky.

**Pull request closed without merging**